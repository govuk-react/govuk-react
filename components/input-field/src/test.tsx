import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';

import { InputField } from '.';

describe('InputField', () => {
  const example = 'example';
  const wrapper = <InputField>{example}</InputField>;
  const meta = {
    touched: true,
    error: example,
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <InputField hint={example} meta={meta}>
        {example}
      </InputField>,
      div
    );
  });

  it('should render an input type="text"', () => {
    const output = shallow(wrapper);
    expect(output.find('input[type="text"]')).toBeTruthy();
  });

  it('should render a label', () => {
    const output = shallow(wrapper);
    expect(output.find('label')).toBeTruthy();
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
