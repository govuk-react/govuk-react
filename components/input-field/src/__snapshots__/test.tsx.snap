// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c0:after {
  content: '';
  display: table;
  clear: both;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c2 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  margin-top: 0;
  padding: 5px;
  border: 2px solid #0b0c0c;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin-bottom: 0;
}

.c2:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c2::-webkit-outer-spin-button,
.c2::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c2[type="number"] {
  -moz-appearance: textfield;
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 0;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 0;
  }
}

@media print {
  .c2 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 0;
  }
}

<ForwardRef
  input={Object {}}
  meta={Object {}}
>
  <Label
    error={false}
  >
    <src__StyledLabel
      error={false}
    >
      <label
        className="c0"
      >
        <LabelText>
          <src__StyledLabelText>
            <span
              className="c1"
            >
              example
            </span>
          </src__StyledLabelText>
        </LabelText>
        <ForwardRef
          error={false}
          type="text"
        >
          <src__StyledInput
            error={false}
            type="text"
          >
            <input
              className="c2"
              type="text"
            />
          </src__StyledInput>
        </ForwardRef>
      </label>
    </src__StyledLabel>
  </Label>
</ForwardRef>
`;
